Cannot lazily load views in 0.10.1

I was lazily loading views in 0.9, and some views needed to use the same function, e.g.

```
url('/<lang:lang>/blog/', 'blog.views.blog_index', defaults={'page': 1})
url('/<lang:lang>/blog/page/<int:page>', 'blog.views.blog_index')
```

Upon updating Flask to 0.10.1, I now get the following error:
```
AssertionError: View function mapping is overwriting an existing endpoint function
```

The cause is the same as #796, but [your current documentation still says this is a valid way of having a central URL map](http://flask.pocoo.org/docs/patterns/lazyloading/).

Is there a way around this, beyond just storing every single `LazyView` in a dict and looking it up each time `url()` is called? The above documentation needs to be updated to reflect the answer.

--- a/docs/patterns/lazyloading.rst
+++ b/docs/patterns/lazyloading.rst
@@ -90,10 +90,7 @@ Then you can define your central place to combine the views like this::
 You can further optimize this in terms of amount of keystrokes needed to
 write this by having a function that calls into
 :meth:`~flask.Flask.add_url_rule` by prefixing a string with the project
-name and a dot, and by wrapping `view_func` in a `LazyView` as needed.
-Additionally, instead of just passing in one url_rule at a time, we're going 
-to use a list to hold one to many url_rules. This will allow us to have several
-routes for a single endpoint::
+name and a dot, and by wrapping `view_func` in a `LazyView` as needed.  ::
 
     def url(import_name, url_rules=[], **options):
         view = LazyView('yourapplication.' + import_name)
@@ -101,7 +98,7 @@ routes for a single endpoint::
             app.add_url_rule(url_rule, view_func=view, **options)
 
     # add a single route to the index view
-    url('views.index', ['/']  )
+    url('views.index', ['/'])
 
     # add two routes to a single function endpoint
     url_rules = ['/user/','/user/<username>']

--- a/docs/patterns/lazyloading.rst
+++ b/docs/patterns/lazyloading.rst
@@ -32,8 +32,8 @@ Imagine the current application looks somewhat like this::
     def user(username):
         pass
 
-Then the centralized approach you would have one file with the views
-(`views.py`) but without any decorator::
+Then, with the centralized approach you would have one file with the views
+(:file:`views.py`) but without any decorator::
 
     def index():
         pass
@@ -54,7 +54,7 @@ Loading Late
 ------------
 
 So far we only split up the views and the routing, but the module is still
-loaded upfront.  The trick to actually load the view function as needed.
+loaded upfront.  The trick is to actually load the view function as needed.
 This can be accomplished with a helper class that behaves just like a
 function but internally imports the real function on first use::
 
@@ -90,14 +90,22 @@ Then you can define your central place to combine the views like this::
 You can further optimize this in terms of amount of keystrokes needed to
 write this by having a function that calls into
 :meth:`~flask.Flask.add_url_rule` by prefixing a string with the project
-name and a dot, and by wrapping `view_func` in a `LazyView` as needed::
+name and a dot, and by wrapping `view_func` in a `LazyView` as needed.
+Additionally, instead of just passing in one url_rule at a time, we're going 
+to use a list to hold one to many url_rules. This will allow us to have several
+routes for a single endpoint::
 
-    def url(url_rule, import_name, **options):
+    def url(import_name, url_rules=[], **options):
         view = LazyView('yourapplication.' + import_name)
-        app.add_url_rule(url_rule, view_func=view, **options)
+        for url_rule in url_rules:
+            app.add_url_rule(url_rule, view_func=view, **options)
 
-    url('/', 'views.index')
-    url('/user/<username>', 'views.user')
+    # add a single route to the index view
+    url('views.index', ['/']  )
+
+    # add two routes to a single function endpoint
+    url_rules = ['/user/','/user/<username>']
+    url('views.user', url_rules)
 
 One thing to keep in mind is that before and after request handlers have
 to be in a file that is imported upfront to work properly on the first

